api/handlers.go

```diff
@@ -260,18 +295,23 @@ func (api *API) registerPOST(w http.ResponseWriter, req *http.Request, _ httprou
 		api.WriteError(w, errors.AddContext(err, "failed to validate challenge response"), http.StatusBadRequest)
 		return
 	}
+	// Parse the request's body.
+	var payload credentialsDTO
+	err = json.Unmarshal(body, &payload)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
 	// Validate the email address.
-	email := req.PostFormValue("email")
-	e, err := mail.ParseAddress(email)
+	e, err := mail.ParseAddress(payload.Email)
```

Comment:
`nit` can we use `email`?

Reply:
`e` is a temporary variable that lives for 3 lines. I can change it to `emailAddr` but I would rather not use `email` as that overlaps with a package name (which is the original reason for me to make this change, as seen in this diff).